### DISCUSSION TITLE
BigNumberLine: Add difference to compare lag

### DISCUSSION THREAD

**Discussion by bonedi** [apache/superset#20356] [Ideas]
The Compare Lag of BigNumberLine Chart is calculating the percentage change. In some cases a difference is desired. For example if the number is an percantage allready. So it would be great if you can choose between percentage change and absolute change. 
Currently the code for calulating the change is:
```    if (compareLag > 0) {
      const compareIndex = compareLag;
      if (compareIndex < sortedData.length) {
        const compareValue = sortedData[compareIndex][1];
        // compare values must both be non-nulls
        if (bigNumber !== null && compareValue !== null && compareValue !== 0) {
          percentChange = (bigNumber - compareValue) / Math.abs(compareValue);
          formattedSubheader = `${formatPercentChange(
            percentChange,
          )} ${compareSuffix}`;
        }
      }
    }
```

One way to change it, would be to create a radio button with value percentChangeDesired (=True or False). Then the Code would change to:
``` if (compareLag > 0) {
      const compareIndex = compareLag;
      if (compareIndex < sortedData.length) {
        const compareValue = sortedData[compareIndex][1];
        // compare values must both be non-nulls
        if (bigNumber !== null && compareValue !== null && compareValue !== 0) {
          if (percentChangeDesired){
              percentChange = (bigNumber - compareValue) / Math.abs(compareValue);
              formattedSubheader = `${formatPercentChange(
                percentChange,
              )} ${compareSuffix}`;
          }
          else {
              absoluteChange = bigNumber - compareValue;
              formattedSubheader = `${absoluteChange} ${compareSuffix}`;
         }
        }
      }
    }

```

I have no experiance in TypeScript, so the code won't work 100% for sure. But I think you get an Idea of what I was thinking of.

**Comment by rusackas** [comment]
Since this is more feature request than bug report, I'll go ahead and move this over to a Discussion. I'm curious if @zhaoyongjie has feelings about this idea.

**Comment by zhaoyongjie** [comment]
Hi @boned @rusackas, the time comparison pane(this pane is in the `Advanced Analytics` section) was missing in the `BigNumber/w treadling`, the current comparison was calculated in the frontend. I suggest moving the `time comparison` pane in and removing the frontend calculation. the `time comparison` supports `difference`, `percentage change`, and `ratio`.  The `time comparsion` also interact with `resample` to interpolate missing values.

**Comment by bonedi** [comment]
Hi @zhaoyongjie, will this allow showing the change under the big number? Like in this example:
![image](https://user-images.githubusercontent.com/49313298/173292804-06403b77-6486-4459-a2ae-0a3929e89b5e.png)

**Reply by zhaoyongjie** [reply]
Yes, I think so.
